src/metadata_validator.py

```diff
@@ -149,8 +149,41 @@ def validateNode(self, dataRecord, model):
         
         return STATUS_PASSED, None, None
     
-    def validate_required_props(self, dataRecord, model):
-        return None
+    def validate_required_props(self, data_record, node_definition):
+        result = {"result": ERROR, "errors": [], "warnings": []}
+        # check the correct format from the node_definition
+        if "model" not in node_definition.keys() or "nodes" not in node_definition["model"].keys():
+            result["errors"].append(f"node definition is not correctly formatted.")
```

Comment:
error should type of ErrorMessage,   warning use the same type.   Don't use plain string as error or warning
{ title: String description: String }

Reply:
updated it to error object